### DISCUSSION TITLE
Renderer dropdown on top-right could change desktop and mobile renderers at the same time

### DISCUSSION THREAD

**Discussion by chrisnolet** [godotengine/godot-proposals#7767] [Editor]
**Proposal:**

The renderers dropdown at the top-right of the main editor window is incredibly convenient for switching between renderers. It currently changes the rendering method for desktop users.

With the recent influx of Unity users (myself included), people have been caught off-guard by the current behavior of the dropdown, however. New users expect that setting the renderer using the dropdown will change both `rendering/renderer/rendering_method` and `rendering/renderer/rendering_method.mobile`. In the worst case, new users set the toggle before deploying to mobile, and are disappointed when Godot does not seem capable of rendering at different quality levels.

The toggle is often a 'top line' change: that is, it is used to get a sense of what is possible for a project. Once a developer establishes the level of quality they require for a game/platform, they will either leave all platforms on the same setting, or refine their choice by selecting different renderers for different platforms in the Project Settings.

In the case where multiple renderers are already selected for different platforms, the dropdown could either continue to display the desktop selection by default, or it could display simply 'Rendering Method,' with no option checked. In either case, also adding a 'More Settings...' option that links to the `rendering/renderer` panel could be valuable (TBD).

As mobile VR headsets and phones become more capable, back-and-forth switching for mobile will become more and more commonplace.

**Alternatives considered:**

One argument for the current behavior is that the mobile renderer can always be set via the Project Settings. However, this lacks discoverability and does not take the path of 'least surprise.'

**Comment by theraot** [comment]
Here is an idea: give this setting a prominent position in the Export window.

If the goal is to change a setting when they are going to export for mobile, it does not make sense to set it on the top right corner of the editor. It makes sense to set it where you configure the export for mobile.

And we would want it prominent (perhaps colored the way as the options in the top right corner, and not obscured by scrolling) so people don't miss it when exporting.

As part of the export window, it would be saved in the export presets, however it can default/fallback to whatever is configured in project settings.

*This is just my take, which I submit to the consideration of those interested.*

**Reply by chrisnolet** [reply]
Thanks, @theraot! One point to clarify: I typically deploy to mobile via the toolbar when iterating. (Fourth button from the left in this screenshot.) So I think it makes sense to keep it all together on the toolbar. Thanks for jumping in to the discussion!

<img width="271" alt="Toolbar" src="https://github.com/godotengine/godot-proposals/assets/756383/26cc41b1-fcab-47e2-b0cc-4882e08f35d1">

**Reply by theraot** [reply]
Ah, I see. Yes, it is probably better to always use that one for launching the game from the editor (which is not an export), and that way we avoid confusion with project settings.

Combining that with what I was saying of having it in export presets, now I'm doubting if it is necessary to show it in project settings at all (perhaps the mobile override can be ignored, perhaps the setting can be hidden entirely, as I don't see any situation where it makes sense for it to take precedence... Yet, I suppose it is still necessary to save it there).

---

As alternative, perhaps the UI could go in another direction, and have there discrimination for what settings will be used in the editor, when running in the same device as the editor, and when running in an attached mobile device. As this would mean the editor would not have to restart to make the change... Yet, that would make the UI busier.

---

I wonder what would users prefer in general.

**Comment by Calinou** [comment]
There's already a pull request addressing this: https://github.com/godotengine/godot/pull/72461
